spiderfoot/correlation.py

```diff
@@ -332,6 +418,7 @@ def event_keep(self, event: dict, field: str, patterns: str, patterntype: str) -
                     if value == pattern:
                         return False
                 else:
+                    ret = False
```

Comment:
I'm pretty sure the absence of this line was a bug. Looking at the conditional above fir comparison, the `ret` needs to be initialised. Otherwise, in theory, `ret` could be set to `True` during one iteration of the loop, then fall through here on the next iteration without being reset.